### ISSUE TITLE
stream error after receiving a ping from WhatsApp then a complete disconnection

### ISSUE THREAD

**Issue by alijundi** [kwailo888/AxolotlTestApp#14]
```
Presence: 962799271057@s.whatsapp.net, is: available
DEBUG:
rx <presence from="962799271057@s.whatsapp.net" type="unavailable" last="1457435237"></presence>
DEBUG:
rx <presence from="962799271057@s.whatsapp.net" type="unavailable" last="1457435237"></presence>
Presence: 962799271057@s.whatsapp.net, is: unavailable
DEBUG:
rx <presence from="962799271057@s.whatsapp.net"></presence>
DEBUG:
rx <presence from="962799271057@s.whatsapp.net"></presence>
Presence: 962799271057@s.whatsapp.net, is: available
DEBUG:
rx <presence from="962799271057@s.whatsapp.net" type="unavailable" last="1457435347"></presence>
DEBUG:
rx <presence from="962799271057@s.whatsapp.net" type="unavailable" last="1457435347"></presence>
Presence: 962799271057@s.whatsapp.net, is: unavailable
DEBUG:
rx <presence from="962799271057@s.whatsapp.net"></presence>
DEBUG:
rx <presence from="962799271057@s.whatsapp.net"></presence>
Presence: 962799271057@s.whatsapp.net, is: available
DEBUG:
rx <presence from="962799271057@s.whatsapp.net" type="unavailable" last="1457435617"></presence>
DEBUG:
rx <presence from="962799271057@s.whatsapp.net" type="unavailable" last="1457435617"></presence>
Presence: 962799271057@s.whatsapp.net, is: unavailable
DEBUG:
rx <iq from="s.whatsapp.net" type="get" id="1457435740" xmlns="urn:xmpp:ping"></iq>
DEBUG:
rx <iq from="s.whatsapp.net" type="get" id="1457435740" xmlns="urn:xmpp:ping"></iq>
DEBUG:
rx <stream:error>
rx   <ping id="1457435740"></ping>
rx </stream:error>
DEBUG:
rx <stream:error>
rx   <ping id="1457435740"></ping>
rx </stream:error>
Connection LOST
CALLING WA CONNECT FROM CHECK FUNCTION
Connection LOST
Removed all delegates
FINISHED COLLECTING GARBAGE
Calling ConnectWA!
Connection LOST
Connect Success....
Login success. Next password:
b6JH6j9XXXXXXXXP1BS8rxXY+JfTJc=
Finished Connecting Method -----
Connection Successful
DEBUG:
rx <ib from="s.whatsapp.net">
rx   <offline count="0"></offline>
rx </ib>
DEBUG:
rx <ib from="s.whatsapp.net">
rx   <offline count="0"></offline>
rx </ib>
DEBUG:
rx <presence from="962796239607@s.whatsapp.net"></presence>
DEBUG:
rx <presence from="962796239607@s.whatsapp.net"></presence>
Presence: 962796239607@s.whatsapp.net, is: available
DEBUG:
rx <presence from="962799271057@s.whatsapp.net" type="unavailable" last="1457435617"></presence>
DEBUG:
rx <presence from="962799271057@s.whatsapp.net" type="unavailable" last="1457435617"></presence>
Presence: 962799271057@s.whatsapp.net, is: unavailable
DEBUG:
rx <iq from="s.whatsapp.net" type="error" id="1457435771-149">
rx   <error code="501" text="feature-not-implemented"></error>
rx </iq>
DEBUG:
rx <iq from="s.whatsapp.net" type="error" id="1457435771-149">
rx   <error code="501" text="feature-not-implemented"></error>
rx </iq>
DEBUG:
rx <iq from="962796239607@s.whatsapp.net" type="error" id="1457435771-148">
rx   <error code="404" text="item-not-found"></error>
rx </iq>
DEBUG:
rx <iq from="962796239607@s.whatsapp.net" type="error" id="1457435771-148">
rx   <error code="404" text="item-not-found"></error>
rx </iq>
DEBUG:
rx <iq from="s.whatsapp.net" type="result" id="1457435771-150"></iq>
DEBUG:
rx <iq from="s.whatsapp.net" type="result" id="1457435771-150"></iq>
DEBUG:
rx <presence from="962799271057@s.whatsapp.net"></presence>
DEBUG:
rx <presence from="962799271057@s.whatsapp.net"></presence>
Presence: 962799271057@s.whatsapp.net, is: available

```

**Comment by alijundi** [comment]
what do think @kwailo888

**Comment by wasimjee** [comment]
@alijundi when r u calling ping, show the lines of code

## 

Regards
http://iwr.co.in

**Comment by alijundi** [comment]
@wasimjee for unknown reasons my number got blocked! I am worried now...

**Comment by pablokere** [comment]
Me too!!

Saludos,
Pablo

**Comment by kwailo888** [comment]
You'll have to give me more background on what else your doing for me to help. In my implementation of my code i'm not having any major issues .. blocking numbers can be caused by many different things and is mostly caused by improper usage .. that being said if you could give me your flow of interaction log then i can look to see where the error could be

**Comment by alijundi** [comment]
@kwailo888 I used to send active every once in a while...  and now i am sending ping every once in a while to stay online... in my test environment it is not getting disconnected as before! I am glad.

Now, I am receiving complete messages without the funny characters that were mentioned in issue #12 before. So, there is no way to know when the message is complete or not! @samuelg78 mentioned he was facing this issue also! I am note sure why this is happening.

And for the database connection I am re-implementing your sample code in issue #11 to connect mssql database instead of mysql and using a simpler method.

Also, I am sure you heard about the issue of @oram83 where we are receiving the following during login:

```
The first is:
Whatsapp Error
ID: 1457608166-3
From: s.whatsapp.net
Code: 501
Message Text: feature-not-implemented

Followed by: 
Whatsapp Error
ID: 1457608166-2
From: [my number]@s.whatsapp.net
Code: 404
Message Text: item-not-found
```

This was mentioned in closed issue #16  and I mentioned to @oram83 to remove the listed functions during login:

> Error is related to sendgetprivacylist() and sendgetclientconfig()
> removing those two functions clears the errors!

Do you know why these functions are not implemented any more. I am afraid that the above functions caused one of my numbers to get blocked!
